Client/Frontend/Browser/User Scripts/FarblingProtectionHelper.swift

```diff
@@ -0,0 +1,127 @@
+    "portable-document-format", "document", "doc",
+    "PDF and PS", "com.adobe.pdf", nil
+  ]
+
+  /// Variables representing the suffix of a randomly generated strings used as the plugin name
+  private static let pluginNameThirdParts: [String?] = [
+    "Viewer", "Renderer", "Display", "Plugin",
+    "plug-in", "plug in", "extension", nil
+  ]
+
+  /// A list of fake voice names to be used to generate a fake `SpeechSynthesizer` voice
+  private static let fakeVoiceNames: [String] = [
+    "Hubert", "Vernon", "Rudolph", "Clayton", "Irving",
+    "Wilson", "Alva", "Harley", "Beauregard", "Cleveland",
+    "Cecil", "Reuben", "Sylvester", "Jasper"
+  ]
+
+  static func makeFarblingParams(from randomConfiguration: RandomConfiguration) throws -> String {
+    var generator = ARC4RandomNumberGenerator(seed: randomConfiguration.domainKeyData.getBytes())
+
+    let farblingData = FarblingData(
+      fudgeFactor: Float.random(in: 0.99...1, using: &generator),
+      fakeVoiceName: FarblingProtectionHelper.makeFakeVoiceName(from: &generator),
+      fakePluginData: FarblingProtectionHelper.makeFakePluginData(from: &generator),
+      randomVoiceIndexScale: Float.random(in: 0...1, using: &generator)
+    )
+
+    let encoder = JSONEncoder()
+    let data = try encoder.encode(farblingData)
+    return String(data: data, encoding: .utf8)!
+  }
+
+  /// Generate fake plugin data to be injected into the farbling protection script
+  private static func makeFakePluginData<T: RandomNumberGenerator>(from generator: inout T) -> [FarblingData.FakePluginData] {
+    let pluginCount = Int.random(in: 1...3, using: &generator)
+
+    // Generate 1 to 3 fake plugins
+    return (0..<pluginCount).map { pluginIndex -> FarblingData.FakePluginData in
+      let mimeTypesCount = Int.random(in: 1...3, using: &generator)
+
+      // Generate 1 to 3 fake mime types
+      let mimeTypes = (0..<mimeTypesCount).map { mimeTypeIndex -> FarblingData.FakeMimeTypeData in
+        return FarblingData.FakeMimeTypeData(
+          suffixes: "pdf",
+          type: "application/pdf",
+          description: randomPluginName(from: &generator)
+        )
+      }
+
+      return FarblingData.FakePluginData(
+        name: randomPluginName(from: &generator),
+        filename: "",
+        description: randomPluginName(from: &generator),
+        mimeTypes: mimeTypes
+      )
+    }
+  }
+
+  /// Generate a fake voice name
+  private static func makeFakeVoiceName<T: RandomNumberGenerator>(from generator: inout T) -> String {
+    let fakeName = fakeVoiceNames.randomElement(using: &generator) ?? fakeVoiceNames.first!
+    return fakeName
+  }
+
+  /// Generate a random string using a prefix, middle and suffix where any of those may be empty.
+  /// - Note: May result in an empty string.
+  private static func randomPluginName<T: RandomNumberGenerator>(from generator: inout T) -> String {
+    return [
+      pluginNameFirstParts.randomElement(using: &generator) ?? nil,
```

Comment:
Why unwrap if you will return `nil` as a value anyway?
Remove `?? nil`

Reply:
Sometimes in Swift you get these situations where you get a double optional (i.e. `String??`). You need to unwrap it to nil to get a single optional (i.e. `String?`) But I will move the `?? nil` to the compact map